From b51ec0a2b574f29cde11c26c1e71bb08605b9d12 Mon Sep 17 00:00:00 2001
From: Chirayu Desai <chirayudesai1@gmail.com>
Date: Thu, 19 Jan 2023 02:19:46 +0530
Subject: [PATCH 13/15] Rename microG signature spoofing perm back to upstream
 name

* It doesn't really make a difference so let's just reduce the diff
* Will also help with 1826

Change-Id: I3a459011dc133fb27765d6d722a9daf74be30183
---
 core/res/AndroidManifest.xml                                 | 2 +-
 services/core/java/com/android/server/pm/ComputerEngine.java | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/core/res/AndroidManifest.xml b/core/res/AndroidManifest.xml
index 17637cb101e8..d2729f658ad1 100644
--- a/core/res/AndroidManifest.xml
+++ b/core/res/AndroidManifest.xml
@@ -3579,7 +3579,7 @@
 
     <!-- @hide Allows an application to change the package signature as
          seen by applications. Only allowed to be used by microG.-->
-    <permission android:name="android.permission.MICROG_SPOOF_SIGNATURE"
+    <permission android:name="android.permission.FAKE_PACKAGE_SIGNATURE"
         android:protectionLevel="signature|privileged" />
 
     <!-- @deprecated No longer useful, see
diff --git a/services/core/java/com/android/server/pm/ComputerEngine.java b/services/core/java/com/android/server/pm/ComputerEngine.java
index 44c8cb4dd742..9206ec2d5579 100644
--- a/services/core/java/com/android/server/pm/ComputerEngine.java
+++ b/services/core/java/com/android/server/pm/ComputerEngine.java
@@ -1613,7 +1613,7 @@ public class ComputerEngine implements Computer {
     private PackageInfo mayFakeSignature(AndroidPackage p, PackageInfo pi,
             Set<String> permissions) {
         try {
-            if (permissions.contains("android.permission.MICROG_SPOOF_SIGNATURE")) {
+            if (permissions.contains("android.permission.FAKE_PACKAGE_SIGNATURE")) {
                 if (DEBUG_PACKAGE_INFO) {
                     Log.v(TAG, "Spoofing signature for microG");
                 }
@@ -1621,7 +1621,7 @@ public class ComputerEngine implements Computer {
             }
         } catch (Throwable t) {
             // We should never die because of any failures, this is system code!
-            Log.w("PackageManagerService.MICROG_SPOOF_SIGNATURE", t);
+            Log.w("PackageManagerService.FAKE_PACKAGE_SIGNATURE", t);
         }
         return pi;
     }
-- 
2.25.1

